diff -uNr anaconda-19.31.79__orig/pyanaconda/packaging/yumpayload.py anaconda-19.31.79/pyanaconda/packaging/yumpayload.py
--- anaconda-19.31.79__orig/pyanaconda/packaging/yumpayload.py	2014-04-29 01:45:59.000000000 +0100
+++ anaconda-19.31.79/pyanaconda/packaging/yumpayload.py	2014-07-02 14:36:15.360488588 +0100
@@ -428,7 +428,10 @@
     @property
     def mirrorEnabled(self):
         with _yum_lock:
-            return "fastestmirror" in self._yum.plugins._plugins
+            if productName.startswith("CentOS"):
+                return 0
+            else:
+                return "fastestmirror" in self._yum.plugins._plugins
 
     def getRepo(self, repo_id):
         """ Return the yum repo object. """
