diff -uNrp anaconda-21.48.22.121.orig/pyanaconda/packaging/yumpayload.py anaconda-21.48.22.121/pyanaconda/packaging/yumpayload.py
--- anaconda-21.48.22.121.orig/pyananconda/packaging/yumpayload.py      2017-08-31 16:28:06.672265889 +0000
+++ anaconda-21.48.22.121/pyanaconda/packaging/yumpayload.py    2017-08-31 16:28:38.812826778 +0000
@@ -41,6 +41,7 @@ import sys
 import time
 import hashlib
 from pyanaconda.iutil import execReadlines, ipmi_abort
+from pyanaconda.product import productName
 from pyanaconda.simpleconfig import simple_replace
 from functools import wraps
 from urlgrabber.grabber import URLGrabber, URLGrabError
@@ -519,12 +519,16 @@ reposdir=%s
     @property
     def mirrorEnabled(self):
         with _yum_lock:
-            # yum initializes with plugins disabled, and when plugins are disabled
-            # _yum.plugins is a DummyYumPlugins object, which has no useful attributes.
-            if hasattr(self._yum.plugins, "_plugins"):
-                return "fastestmirror" in self._yum.plugins._plugins
+            # we just skip this on CentOS since we cant support it yet
+            if productName.startswith("CentOS"):
+              return False
             else:
-                return False
+              # yum initializes with plugins disabled, and when plugins are disabled
+              # _yum.plugins is a DummyYumPlugins object, which has no useful attributes.
+              if hasattr(self._yum.plugins, "_plugins"):
+                  return "fastestmirror" in self._yum.plugins._plugins
+              else:
+                  return False

     def getRepo(self, repo_id):
         """Return the yum repo object."""
